src/effects/effectchain.cpp

```diff
@@ -150,38 +154,70 @@ void EffectChain::setEnabled(bool enabled) {
     emit(enabledChanged(enabled));
 }
 
-void EffectChain::enableForChannel(const ChannelHandleAndGroup& handle_group) {
-    if (!m_bAddedToEngine) {
+void EffectChain::enableForInputChannel(const ChannelHandleAndGroup& handle_group) {
```

Comment:
I think it should be no deal to move this to EffectChainSlot. If I follow the call tree, it is called from there or in combination with EffectChainSlot calls anyway.

Reply:
It is a big deal because methods that rely on m_enabledInputChannels are used by EffectChain::clone, which is used by the EffectRack::loadNextChain and EffectRack::loadPrevChain methods which depend on the ability to have EffectChains exist disconnected from being loaded to the engine. We could break EffectRack::loadNextChain and EffectRack::loadPrevChain considering they are nonsensical without the ability to actually save and restore chain presets. It also does not make sense that those are methods of EffectRack and not EffectChainSlot. IMO there is no pressing need to do this now and we should save it for 2.2.

Reply:
Ok